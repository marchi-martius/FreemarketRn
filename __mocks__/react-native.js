const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

module.exports = rn;
