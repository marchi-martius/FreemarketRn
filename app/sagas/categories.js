import { put, call, takeLatest } from 'redux-saga/effects';
import { Actions } from 'react-native-router-flux';

import {
  categoriesSuccess,
  categoriesError,
  requestCategories,
} from '../reducers/categories';

import { fetchAllCategories } from '../api/rest';
import {
  objectWithRecordsToArray,
  isUnauthorizedError,
} from './util';

export function* fetchCategories() {
  try {
    const { data } = yield call(fetchAllCategories);
    const categories = objectWithRecordsToArray(data);
    yield put(categoriesSuccess(categories));
  } catch (e) {
    yield put(categoriesError(e));
    if (isUnauthorizedError(e)) {
      yield call(Actions.SignIn);
    }
  }
}

export function* watchCategoriesRequest() {
  yield takeLatest(requestCategories.getType(), fetchCategories);
}
