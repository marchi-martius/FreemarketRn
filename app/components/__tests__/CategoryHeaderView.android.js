import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CategoryHeaderView from '../CategoryHeaderView.android';
import categories from '../../constants/categories';

it('renders correctly', () => {
  const tree = renderer.create(
    <CategoryHeaderView
      category={categories[0]}
      onIconClicked={() => {}}
      onActionSelected={() => {}}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
