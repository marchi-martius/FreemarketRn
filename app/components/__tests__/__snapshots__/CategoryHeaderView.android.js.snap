// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Icon.ToolbarAndroid
  actions={
    Array [
      Object {
        "iconName": "plus",
        "show": "always",
        "title": "Add Product",
      },
    ]
  }
  navIconName="bars"
  onActionSelected={[Function]}
  onIconClicked={[Function]}
  style={
    Object {
      "backgroundColor": "dodgerblue",
      "height": 56,
    }
  }
  title="Antiques"
  titleColor="white"
/>
`;
